OpenSim/Simulation/SimbodyEngine/Joint.h

```diff
@@ -33,6 +33,25 @@ namespace OpenSim {
 
 class Model;
 
+class JointCannotJoinTheSamePhysicalFrame : public Exception {
+public:
+    JointCannotJoinTheSamePhysicalFrame(const std::string& file,
+        size_t line,
+        const std::string& func,
+        const std::string& thisName,
+        const std::string& parentName,
+        const std::string& childName,
+        const std::string& baseName) :
+        Exception(file, line, func) {
+        std::string msg = "Joint '" + thisName + "' cannot connect '" +
+            parentName + "' to '" + childName +
+            "' because they have the same base PhysicalFrame '" + 
```

Comment:
"they" -> "both frames" is potentially misleading. "Joint A cannot connect B to C because both frames have the same base PhysicalFrame" implies that both B and C have the same base PhysicalFrame as some unspecified third thing. You could say "...because these two frames have..." if the objective is to insert the word "frames" so it's clear what the Joint is trying to connect.